src/mscorlib/src/System/IO/StreamHelpers.CopyValidation.cs

```diff
@@ -20,25 +20,23 @@ public static void ValidateCopyToArgs(Stream source, Stream destination, int buf
                 throw new ArgumentOutOfRangeException(nameof(bufferSize), bufferSize, Environment.GetResourceString("ArgumentOutOfRange_NeedPosNum"));
             }
 
-            bool sourceCanRead = source.CanRead;
-            if (!sourceCanRead && !source.CanWrite)
+            if (!source.CanRead)
             {
-                throw new ObjectDisposedException(null, Environment.GetResourceString("ObjectDisposed_StreamClosed"));
-            }
-
-            bool destinationCanWrite = destination.CanWrite;
-            if (!destination.CanRead && !destinationCanWrite)
-            {
-                throw new ObjectDisposedException(nameof(destination), Environment.GetResourceString("ObjectDisposed_StreamClosed"));
-            }
+                if (!source.CanWrite)
+                {
+                    throw new ObjectDisposedException(null, Environment.GetResourceString("ObjectDisposed_StreamClosed"));
```

Comment:
Swapping the order in the `&&` would have the same effect, but I thought it was worth emphasizing the order of the calls and the fact that the dependent calls are only needed in exceptional cases.